### PR TITLE
Prevent name overwritting of Product Variants when Updating Product Types

### PR DESCRIPTION
**Pull request by teddyondieki**
I want to merge this change because it prevent existing names of ProductVariants from being overwritten during updates on the Product Type page. Issue #15318

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

Docs are not required

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Review by IKarbowiak — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Review by korycins — DISMISSED** [review]

**Comment by teddyondieki** [comment]
@IKarbowiak @korycins I don't have write access, please merge this when you can. Thanks

**Review by zedzior — DISMISSED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by zedzior — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Pull request merged**
